lib/IRGen/GenDecl.cpp

```diff
@@ -2548,8 +2548,10 @@ void IRGenModule::emitGlobalDecl(Decl *D) {
     return;
 
   case DeclKind::MacroExpansion:
-    for (auto *rewritten : cast<MacroExpansionDecl>(D)->getRewritten())
-      emitGlobalDecl(rewritten);
+    if (auto *rewritten = cast<MacroExpansionDecl>(D)->getRewritten())
+      for (auto node : rewritten->getElements())
+        if (auto *decl = node.dyn_cast<Decl *>())
+          emitGlobalDecl(decl);
```

Comment:
We have so many copies of this code to walk the rewritten declarations. As we get other kinds of declaration-generating macros (peer and member, especially), we'll have a lot more duplication for that. I'm concerned that it's far too easy to miss a spot.

Can we introduce requests for these walks, such that the requests ensure that all of the right macros get expanded, and then move this code over to those requests that "flatten" the list of declarations? We're going to be using that over and over.